pkg/descheduler/strategies/pod_lifetime_test.go

```diff
@@ -245,12 +253,34 @@ func TestPodLifeTime(t *testing.T) {
 			pods:                    []v1.Pod{*p12, *p13},
 			nodes:                   []*v1.Node{node1},
 			expectedEvictedPodCount: 1,
+			testFunc:                func(podList []v1.Pod) {},
+		},
+		{
+			description: "Two old pods with different labels, 1 selected by labelSelector, no pod should be evicted since pod terminating",
+			strategy: api.DeschedulerStrategy{
+				Enabled: true,
+				Params: &api.StrategyParameters{
+					PodLifeTime: &api.PodLifeTime{MaxPodLifeTimeSeconds: &maxLifeTime},
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"foo": "bar"},
+					},
+				},
+			},
+			maxPodsToEvictPerNode:   5,
+			pods:                    []v1.Pod{*p12, *p13},
+			nodes:                   []*v1.Node{node1},
+			expectedEvictedPodCount: 1,
+			testFunc: func(podList []v1.Pod) {
+				for _, pod := range podList {
+					pod.DeletionTimestamp = &metav1.Time{}
+				}
+			},
 		},
 	}
 
 	for _, tc := range testCases {
 		fakeClient := &fake.Clientset{}
-
+		tc.testFunc([]v1.Pod{*p12, *p13})
```

Comment:
this is again calling testFunc for just p12 and p13 on every test, even though the pods are only used in 1 test. this shouldn't be necessary

Reply:
Already remove `testFunc`, may I click `Resolve conversation`?